app/controllers/stages_controller.rb

```diff
@@ -35,6 +36,29 @@ def update
     end
   end
 
+  def index
+    tournament = Tournament.find(params[:tournament_id])
+    level_param = index_params[:level]
+    if level_param
+      if level_param == 'current'
+        # TODO: find current stage
+      elsif level_param == 'group'
+        group_stage = tournament.stages.find_by(level: -1)
+        if group_stage
+          render json: group_stage
+        else
+          render json: { error: 'There\'s no group stage for this tournament' }, status: :not_found
+        end
+      elsif level_param.to_i.to_s == level_param
```

Comment:
What about `.is_a? Integer`

Reply:
If Rails doesn't do any auto-conversion magic, `level_param` is always a `String`... but maybe it does, will try